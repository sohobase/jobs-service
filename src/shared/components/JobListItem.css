@import "../styles/colors.css";
@import "../styles/fonts.css";
@import "../styles/variables.css";

:root {
  --item-background-hover: color(white blackness(2.5%));
  --item-label-font-size: calc(var(--font-size-tiny) * 0.85);
  --item-label-offset: calc(var(--unit) * 0.25) calc(var(--unit) * 0.25);
}

.joblistitem {
  position: relative;
  display: flex;
  flex-direction: column;
  padding: var(--offset);
  color: var(--color-text);
  cursor: pointer;
  border-bottom: solid 1px var(--color-divider);

  @apply --box;

  /* -- children */
  & .new {
    display: none;
    position: absolute;
    top: var(--offset);
    left: calc(var(--offset) * -0.5);
    padding: var(--item-label-offset);
    font-size: var(--item-label-font-size);
    font-weight: var(--font-weight-extra-bold);
    background-color: var(--color-accent);
    color: white;
  }

  & .image {
    position: absolute;
    left: calc((var(--offset) * -1));

    @apply --avatar;
  }

  & .top {
    text-transform: uppercase;
    display: flex;

    & .category {
      color: var(--color-text-secondary);
      margin: 0 var(--offset);
      flex-grow: 1;

      &:hover {
        color: var(--color-accent);
      }
    }

    & .company {
    }
  }

  & .position {
    font-size: var(--font-size-normal);
    margin: calc(var(--offset) * 0.5) 0;
  }

  & .about {
    height: calc(var(--font-size-normal) * 2);
    overflow: hidden;
    text-overflow: ellipsis;
  }

  & .top,
  & .about {
    color: var(--color-text-secondary);
    font-size: var(--font-size-small);
  }

  /* -- behaviors */
  &:hover {
    background-color: var(--item-background-hover);

    & .position {
      color: var(--color-accent);
    }
  }
}
