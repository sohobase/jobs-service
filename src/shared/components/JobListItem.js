import classnames from 'classnames';
import React, { PropTypes } from 'react';
import { Link } from 'react-router';
import { ProviderDate } from '../providers';
import style from './JobListItem.css';

const JobListItem = (props = {}) => {
  const {
    id,
    category,
    createdAt = (new Date()),
    company,
    companyImage,
    position,
  } = props;

  const isNew = ProviderDate.inLast24Hours(createdAt);
  const classes = classnames(style.joblistitem, style[category]);

  return (
    <Link to={`/job/${id}`}>
      <li className={classes}>
        { isNew && <abbr className={style.new}>NEW</abbr> }
        { companyImage && <img src={companyImage} alt={company} className={style.image} /> }
        <div className={style.info}>
          <small className={style.company}>{company}</small>
          <strong className={style.position}>{position}</strong>
        </div>
        <small className={style.date}>{ProviderDate.ago(createdAt)}</small>
      </li>
    </Link>
  );
};

JobListItem.propTypes = {
  category: PropTypes.string,
  company: PropTypes.string,
  companyImage: PropTypes.string,
  id: PropTypes.string,
  position: PropTypes.string,
  createdAt: PropTypes.any,
};

export default JobListItem;
