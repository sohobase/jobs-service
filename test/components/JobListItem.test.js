import React from 'react';
import renderer from 'react-test-renderer';

import JobListItem from '../../src/shared/components/JobListItem';

describe('<JobListItem />', () => {
  // -- Renders
  it('renders by default', () => {
    const tree = renderer.create(<JobListItem />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a name property', () => {
    const tree = renderer.create(<JobListItem name="Test job" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
