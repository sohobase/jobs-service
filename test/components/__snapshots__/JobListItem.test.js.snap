exports[`<JobListItem /> renders a name property 1`] = `
<a
  onClick={[Function]}
  style={Object {}}>
  <li
    className="joblistitem undefined">
    <abbr
      className="new">
      NEW
    </abbr>
    <div
      className="info">
      <small
        className="company" />
      <strong
        className="position" />
    </div>
    <small
      className="date">
      a few seconds ago
    </small>
  </li>
</a>
`;

exports[`<JobListItem /> renders by default 1`] = `
<a
  onClick={[Function]}
  style={Object {}}>
  <li
    className="joblistitem undefined">
    <abbr
      className="new">
      NEW
    </abbr>
    <div
      className="info">
      <small
        className="company" />
      <strong
        className="position" />
    </div>
    <small
      className="date">
      a few seconds ago
    </small>
  </li>
</a>
`;
